kernel/packages/unity-interface/nativeMessagesBridge.ts

```diff
@@ -212,6 +212,9 @@ export class NativeMessagesBridge {
   }
 
   entityComponentCreateOrUpdate(payload: UpdateEntityComponentPayload) {
+    if (payload.json.length > 49000) {
```

Comment:
This check shouldn't be for any `json` payload that has creator inputs?

This error may be triggered by:
- `callSharedComponentUpdate`
- `callOpenNftDialog`
- `callOpenExternalUrl`

Reply:
Also, this check should be on `SceneRuntime.ts` for improved performance (as it would run in the scene worker).

We can implement it like this:

```
        updateEntityComponent(entityId: string, componentName: string, classId: number, json: string): void {
          if (json.length > 49000) {
              return;
          }
          
          if (componentNameRE.test(componentName)) {
            that.events.push({
              type: 'UpdateEntityComponent',
              tag: sceneId + '_' + entityId + '_' + classId,
              payload: {
                entityId,
                classId,
                name: componentName.replace(componentNameRE, ''),
                json: generatePBObject(classId, json)
              } as UpdateEntityComponentPayload
            })
          }
```

And so on for the remaining message types.

Reply:
I applied this, would you mind taking a look again? I also couldn't find anything related to `callSharedComponentUpdate` in `SceneRuntime`. Could you point me in the right direction here?

Reply:
`callSharedComponentUpdate` is `componentUpdated`. The event names are messed up from kernel side.